ntoskrnl/config/cmapi.c

```diff
@@ -2128,8 +2128,12 @@ CmLoadKey(IN POBJECT_ATTRIBUTES TargetKey,
     else
     {
         DPRINT1("CmpLinkHiveToMaster failed, Status %lx\n", Status);
-        /* FIXME: TODO */
-        // ASSERT(FALSE); see CORE-17263
+
+        /* Close file handles of this hive */
+        CmpCloseHiveFiles(CmHive);
+
+        /* Perform a cleanup against this hive */
+        CmpDestroyHive(CmHive);
         ExReleasePushLock(&CmpLoadHiveLock);
```

Comment:
While here, could merge identical line in both cases.

Reply:
I'd say this is a legit comment. The lock is acquired outside the condition, and should be released outside it too.